### DISCUSSION TITLE
Conditionnaly cancel query Middleware

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#13506] [Q&A]
Hello, 

I am currently learning how middleware works. 
https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware

Is there a possibility to cancel an operation if a check occuring in the middleware did not work ? 
```
prisma.$use(async (params, next) => {

  const result = await next(params)
  // if(condition) { revert to initial situation)
  return result
})

```

**Comment by nurul3101** [comment]
Hey @laurent512 👋 
You could use $disconnect to cancel the query but please note that the database connection will be terminated on invoking [disconnect](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#disconnect).
When the next query is invoked the database connection would be established lazily, so you won't need to explicitly call [$connect](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#connect).

**Reply by laurent512** [reply]
Sorry it is not clear to me how this does work. 
for example, I would expect this to cancel all query and it does not .
```
prisma.$use(async (params, next) => {
  await prisma.$disconnect();
   const result = await next(params);
  return result; }
```

**Comment by janpio** [comment]
Yes, just return the middleware and do not execute `await next` - that is where you run the actual query.
If you have to run `await next` to actually know the condition, you will probably need to manually undo all the changes some way before you return the middleware.

**Reply by vladi-stevanovic** [reply]
👋 Hello @laurent512  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
